### PR TITLE
CARTO: API v3.3 reenable

### PR DESCRIPTION
**Pull request by felixpalmer**
For https://github.com/visgl/deck.gl/issues/8148

#### Change List
- Use `spatialDataColumn` & `spatialDataType` URL parameters
- Use `'geo'` for vector sources
- Defaults for `spatialDataColumn`

### CODE REVIEW

File: modules/carto/src/sources/quadbin-table-source.ts [hunk 1]
```diff
@@ -19,19 +26,22 @@ export const quadbinTableSource = async function (
...
+    spatialDataColumn = 'quadbin',
     tableName
   } = options;
-  const urlParameters: UrlParameters = {aggregationExp, name: tableName};
+
+  const urlParameters: UrlParameters = {
+    aggregationExp,
+    name: tableName,
+    spatialDataColumn,
+    spatialDataType: 'quadbin'
```
Comment (alasarr) [inline modules/carto/src/sources/quadbin-table-source.ts:37]: It should use the type SpatialDataType.quadbin

File: modules/carto/src/sources/quadbin-query-source.ts [hunk 2]
```diff
@@ -19,17 +26,19 @@ export const quadbinQuerySource = async function (
...
-    spatialDataColumn = 'quadbin:quadbin',
+    spatialDataColumn = 'quadbin',
     queryParameters
   } = options;
-  const urlParameters: UrlParameters = {aggregationExp, q: sqlQuery};
+  const urlParameters: UrlParameters = {
+    aggregationExp,
+    q: sqlQuery,
+    spatialDataColumn,
+    spatialDataType: 'quadbin'
```
Comment (alasarr) [inline modules/carto/src/sources/quadbin-query-source.ts:36]: Use the type

File: modules/carto/src/sources/h3-query-source.ts [hunk 3]
```diff
@@ -18,17 +25,19 @@ export const h3QuerySource = async function (
...
     sqlQuery,
-    spatialDataColumn = 'h3:h3',
+    spatialDataColumn = 'h3',
     queryParameters
   } = options;
-  const urlParameters: UrlParameters = {aggregationExp, q: sqlQuery};
+  const urlParameters: UrlParameters = {
+    aggregationExp,
+    spatialDataColumn,
+    spatialDataType: 'h3',
```
Comment (alasarr) [inline modules/carto/src/sources/h3-query-source.ts:34]: Use the type

**Review by alasarr — CHANGES_REQUESTED** [review]

File: modules/carto/src/sources/quadbin-query-source.ts (hunk 2 above)
Comment (felixpalmer) [reply modules/carto/src/sources/quadbin-query-source.ts:36]: Actually we are moving towards just using the strings in the deck/loaders codebase. It makes it easier to read as you don't have to go and look at what the constant is and TypeScript ensures that it is valid anyway.

Comment (alasarr) [reply modules/carto/src/sources/quadbin-query-source.ts:36]: ok

**Review by alasarr — APPROVED** [review]

**Pull request merged**